### DISCUSSION TITLE
How do I use addons in the turbowarp project player?

### DISCUSSION THREAD

**Discussion by fractioneater** [ScratchAddons/ScratchAddons#5213] [Help] [answered]
I have the TurboWarp button addon enabled, and set to replace player. In the addon settings, there is a toggle that says "enable addons." I've tried putting ?hqpen at the end of the URL, (and some other things), but it doesn't seem to have any effect.

How do I use addons in the turbowarp embedded project player?

**Comment by CST1229** [comment, marked as answer]
Addons are not advanced settings; addons refer to things like Gamepad support. I don't think you can use advanced settings using the addon (other than setting the button to open the page in a new tab or by storing advanced settings in the project).

**Reply by fractioneater** [reply]
Oh, that makes sense.

**Reply by Samq64** [reply]
Actually it is possible: https://github.com/ScratchAddons/ScratchAddons/issues/4005#issuecomment-1252433168
> The connected PR is stalled so I believe it may be worth reiterating that there is a URL parameter for an in-embed settings screen: https://docs.turbowarp.org/embedding#settings-button

**Reply by fractioneater** [reply]
So it's possible, it just hasn't been implemented in Scratch Addons yet. I might open an issue requesting this.

**Reply by fractioneater** [reply]
Nevermind, I'm stupid, you just sent a link to an issue about it.

**Reply by Samq64** [reply]
There are a couple other TurboWarp button related PRs open, but I think I'll just PR that solution anyway.

**Reply by Samq64** [reply]
nvm, Garbo already did.
